File: 백준/Bronze/2506. 점수계산/점수계산.cc
#include <iostream>
using namespace std;

int main() {
	
	int n;
	cin >> n;

	int sum = 0;
	int plus = 0;
	for (int i = 0; i < n; i++)
	{
		int answer;
		cin >> answer;

		if (answer == 0)
		{
			plus = 0;
		}

		if (answer == 1)
		{
			sum += ++plus;
		}
	}

	cout << sum << endl;
}

File: 백준/Bronze/2506. 점수계산/점수계산.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();

        int sum = 0;
        int plus = 0;
        for (int i = 0; i < n; i++) {
            int answer = sc.nextInt();

            if (answer == 0) {
                plus = 0;
            } else {
                sum += ++plus;
            }
        }

        System.out.println(sum);
    }
}